src/main/java/com/epam/ta/reportportal/core/filter/IShareUserFilterHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.epam.ta.reportportal.core.filter;
+
+import com.epam.ta.reportportal.auth.ReportPortalUser;
+import com.epam.ta.reportportal.commons.querygen.Filter;
+import com.epam.ta.reportportal.ws.model.filter.UserFilterResource;
+import org.springframework.data.domain.Pageable;
+
+public interface IShareUserFilterHandler {
```

Comment:
is that possible to create one generic interface for all sharable entities (filters, dashboards,widgets)?

Reply:
Any object can be shared. There is no special requirements for these objects except they should have "getId()" method and this method already exist in  com.epam.ta.reportportal.entity.filter.Queryable